### PR TITLE
Solves [Bug] #595  Setting TabView.SelectedIndex does not "visually" switch tabs

### LINKED ISSUE #595: [Bug] Setting TabView.SelectedIndex does not "visually" switch tabs
### Description
If I programmatically set the `SelectedIndex` of a `TabView` to say 1 when it is originally on `SelectedIndex` of 0, it appears that internally the new Tab is selected but visually the the `TabView` appears to still be on Tab 0. If you were then to physically tap on Tab 1, nothing would happen since it is internally already on Tab 1. Tapping on Tab 0 and then Tab 1 will finally successfully show Tab 1. See gif below:

### Steps to Reproduce

XAML:
```xml
<?xml version="1.0" encoding="utf-8" ?>
<ContentPage xmlns="http://xamarin.com/schemas/2014/forms"
             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
			 xmlns:xct="http://xamarin.com/schemas/2020/toolkit"
             x:Class="App4.MainPage">

	<StackLayout>
		<Button Text="Change Tab"
                              Clicked="Button_Clicked" />
		<xct:TabView x:Name="MyTabView"
					 TabIndicatorPlacement="Bottom"
					 TabIndicatorColor="Orange">
			<xct:TabViewItem Text="First">
				<Label Text="First Tab" />
			</xct:TabViewItem>
			<xct:TabViewItem Text="Second">
				<Label Text="Second Tab" />
			</xct:TabViewItem>
		</xct:TabView>
	</StackLayout>
</ContentPage>
```

Code Behind:
```csharp
namespace App4
{
	public partial class MainPage : ContentPage
	{
		public MainPage()
		{
			InitializeComponent();
		}

		private void Button_Clicked(object sender, EventArgs e)
		{
			MyTabView.SelectedIndex = 1;
		}
	}
}
```

### Expected Behavior
Upon button click, the `TabView` should visually navigate to the next Tab on `SelectedIndex` change.
### Actual Behavior
tabview

### Basic Information

- Version with issue: 1.0.0-pre5
- IDE:
- Platform Target Frameworks: 
  - iOS:  14.2
  - Android: 10.0

### Workaround
```csharp
MethodInfo dynMethod = MyTabView.GetType().GetMethod("UpdateSelectedIndex", BindingFlags.NonPublic | BindingFlags.Instance);
dynMethod?.Invoke(MyTabView, new object[] { 1, false });
```

### PR DESCRIPTION
**Pull request by Jouna77**
Setting TabView.SelectedIndex does not "visually" switch tabs

### Description of Change ###
 Solves setting TabView.SelectedIndex does not "visually" switch tabs
Adding oldPosition value to UpdateSelectedIndex function in UpdateSelectedIndex avoids the false true on
` if (oldPosition == newPosition)`

### Bugs Fixed ###
https://github.com/xamarin/XamarinCommunityToolkit/issues/595
- Fixes #595

### API Changes ###

None

Added: 
- Parameter `oldPosition ` on `TabView.shared.UpdateSelectedIndex `

### PR Checklist ###

- [ ] Has tests (if omitted, state reason in description)
- [ ] Has samples (if omitted, state reason in description)
- [ ] Rebased on top of main at time of PR
- [ ] Changes adhere to coding standard
- [ ] Updated [documentation](https://github.com/MicrosoftDocs/xamarin-communitytoolkit)

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/xamarin/XamarinCommunityToolkit?pullRequest=738) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/xamarin/XamarinCommunityToolkit?pullRequest=738) before we can accept your contribution.<br/><br/>:x: Jouna77 [sign now](https://cla.dotnetfoundation.org/xamarin/XamarinCommunityToolkit?pullRequest=738)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/xamarin/XamarinCommunityToolkit?pullRequest=738) it.</sub>

**Review by jsuarezruiz — APPROVED** [review]

File: src/CommunityToolkit/Xamarin.CommunityToolkit/Views/TabView/TabView.shared.cs [hunk 1]
```diff
@@ -745,12 +745,10 @@ bool CanUpdateSelectedIndex(int selectedIndex)
 			return true;
 		}
 
-		void UpdateSelectedIndex(int position, bool hasCurrentItem = false)
+		void UpdateSelectedIndex(int oldPosition, int position, bool hasCurrentItem = false)
```
Comment (jsuarezruiz) [inline src/CommunityToolkit/Xamarin.CommunityToolkit/Views/TabView/TabView.shared.cs:748]: Is it simpler perhaps, to validate the old and new index before invoking UpdateSelectedIndex?. Could you add an example where validate the fix?

**Review by jsuarezruiz — CHANGES_REQUESTED** [review]

**Review by jfversluis — APPROVED** [review]

**Comment by jfversluis** [comment]
I think this is now solved by #903

**Comment by jfversluis** [comment]
Maybe I was too quick on this 😬

**Comment by Jouna77** [comment]
Example forked and updated from: @MauricioCodeguez
https://github.com/Jouna77/TabViewDemo

**Review by jfversluis — APPROVED** [review]

**Review by jfversluis — APPROVED** [review]

**Pull request merged**